presto-hive/src/test/java/com/facebook/presto/hive/TestHiveCanonicalPlanHashes.java

```diff
@@ -0,0 +1,147 @@
+    {
+        QueryRunner queryRunner = getQueryRunner();
+
+        try {
+            queryRunner.execute("CREATE TABLE test_orders WITH (partitioned_by = ARRAY['ds', 'ts']) AS " +
+                    "SELECT orderkey, orderpriority, comment, custkey, '2020-09-01' as ds, '00:01' as ts FROM orders WHERE orderkey < 1000 " +
+                    "UNION ALL " +
+                    "SELECT orderkey, orderpriority, comment, custkey, '2020-09-02' as ds, '00:02' as ts FROM orders WHERE orderkey < 1000");
+
+            assertSamePlanHashes(
+                    "SELECT orderkey from test_orders",
+                    "SELECT orderkey from test_orders",
+                    CONNECTOR);
+            assertSamePlanHashes(
+                    "SELECT orderkey from test_orders where ds > '2020-09-01'",
+                    "SELECT orderkey from test_orders where ds = '2020-09-02'",
+                    CONNECTOR);
+            assertSamePlanHashes(
+                    "SELECT orderkey from test_orders where ds = '2020-09-01' AND orderkey < 10 AND ts >= '00:01'",
+                    "SELECT orderkey from test_orders where ds = '2020-09-02' AND orderkey < 10 AND ts >= '00:02'",
+                    CONNECTOR);
+
+            assertDifferentPlanHashes(
+                    "SELECT orderkey from test_orders where ds = '2020-09-01' AND orderkey < 10",
+                    "SELECT orderkey from test_orders where ds = '2020-09-02' AND orderkey < 20",
+                    CONNECTOR);
+
+            assertSamePlanHashes(
+                    "SELECT orderkey, CAST(1 AS VARCHAR) from test_orders where ds = '2020-09-01' AND orderkey < 10",
+                    "SELECT orderkey, CAST(2 AS VARCHAR) from test_orders where ds = '2020-09-02' AND orderkey < 10",
+                    REMOVE_SAFE_CONSTANTS);
+            assertDifferentPlanHashes(
+                    "SELECT orderkey, CAST(1 AS VARCHAR) from test_orders where ds = '2020-09-01' AND orderkey < 10",
+                    "SELECT orderkey, CAST(1 AS VARCHAR) from test_orders where ds = '2020-09-02' AND orderkey < 20",
+                    REMOVE_SAFE_CONSTANTS);
+        }
+        finally {
+            queryRunner.execute("DROP TABLE IF EXISTS test_orders");
+        }
+    }
+
+    private void assertSamePlanHashes(String sql1, String sql2, PlanCanonicalizationStrategy strategy)
+    {
+        List<String> hashes1 = getPlanHashes(sql1, strategy);
+        List<String> hashes2 = getPlanHashes(sql2, strategy);
+        assertEquals(hashes1, hashes2);
+    }
+
+    private void assertDifferentPlanHashes(String sql1, String sql2, PlanCanonicalizationStrategy strategy)
+    {
+        List<String> hashes1 = getPlanHashes(sql1, strategy);
+        List<String> hashes2 = getPlanHashes(sql2, strategy);
+        assertNotEquals(hashes1, hashes2);
+    }
+
+    private List<String> getPlanHashes(String sql, PlanCanonicalizationStrategy strategy)
+    {
+        Session session = pushdownFilterEnabled();
```

Comment:
why is pushdownFilterEnabled needed?

Reply:
So partition information is pushed to table layout. Also resembles prod settings.